### PR TITLE
core: remove globals from `externs.d.ts`

### PR DESCRIPTION
**Pull request by adamraine**
https://github.com/GoogleChrome/lighthouse/pull/14441#discussion_r992749543

- Global util types (`SelfMap`, `Immutable`, etc) are moved to `util.d.ts` and are accessible via `LH.Util`
- Global namespace modifications (`Intl`, `Window`) are moved to `node.d.ts`
- Any remaining external types (`Flags`, `RunnerResult`, etc) are staying in `externs.d.ts`

This is to separate global definitions only useful to us and non-global definitions we eventually want to expose.

### CODE REVIEW

**Comment by brendankenny** [comment]
> Global util types (SelfMap, Immutable, etc) are moved to util.d.ts and are accessible via LH.Util

These seem like a prime subject for real imports instead of being in a new `LH` namespace. They're mostly used in d.ts files and they've never been namespaced before so no need to update existing usage. Some of the uses of them aren't needed as well...e.g. `LH.Util.Immutable<LH.Config.Settings>` should really just be `LH.Audit.Context['settings']` in a lot (all?) of the updated cases.

File: types/node.d.ts [hunk 1]
```diff
@@ -7,6 +7,36 @@
...
   var isDevtools: boolean | undefined;
   var isLightrider: boolean | undefined;
+
+  // Augment Intl to include
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales
+  namespace Intl {
+    var getCanonicalLocales: (locales?: string | Array<string>) => Array<string>;
+  }
+
+  interface Window {
```
Comment (brendankenny) [inline types/node.d.ts:17]: this is really weird in a file `node.d.ts` at first glance :). Maybe deserves a comment

**Comment by adamraine** [comment]
> They're mostly used in d.ts files and they've never been namespaced before so no need to update existing usage.

Most of the usages I had to update were in `.js` files. Also importing these directly with jsdoc requires a template:

```js
/**
 * @template T
 * @typedef {import('../types/util.js').default.SelfMap<T>} SelfMap
 */
```

> e.g. LH.Util.Immutable<LH.Config.Settings> should really just be LH.Audit.Context['settings'] in a lot (all?) of the updated cases.

I'll try and update as many of these as I can.

File: types/util.d.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/**
```
Comment (brendankenny) [inline types/util.d.ts:1]: wdyt about `types/utility-types.d.ts`? Even if redundant it self describes a little better and differentiates more from the four `util.*` files we already have

Comment (adamraine) [reply types/util.d.ts:1]: Sure but still exposed as `LH.Util`?

**Pull request merged**